OpenRA.Mods.Common/Widgets/ProductionTabsWidget.cs

```diff
@@ -200,36 +189,33 @@ void Scroll(int amount)
 			listOffset = Math.Min(0, Math.Max(Bounds.Width - rightButtonRect.Width - leftButtonRect.Width - contentWidth, listOffset));
 		}
 
-		// Is added to world.ActorAdded by the SidebarLogic handler
-		public void ActorChanged(Actor a)
+		void Update()
 		{
-			if (a.Info.HasTraitInfo<ProductionQueueInfo>())
-			{
-				var allQueues = a.World.ActorsWithTrait<ProductionQueue>()
-					.Where(p => p.Actor.Owner == p.Actor.World.LocalPlayer && p.Actor.IsInWorld && p.Trait.Enabled)
-					.Select(p => p.Trait).ToList();
+			var allQueues = world.ActorsWithTrait<ProductionQueue>()
```

Comment:
Running this every tick isn't great for performance.  It should be possible to keep most of the original code to maintain the list of "all queues of my actors in the world" and then filter that down to "all *enabled* queues of my actors in the world" when needed each tick, avoiding most of the overhead.